lib/repl.js

```diff
@@ -602,7 +602,19 @@ REPLServer.prototype.complete = function(line, callback) {
           // if (e) console.log(e);
 
           if (obj != null) {
-            if (util.isObject(obj) || util.isFunction(obj)) {
+            if (Array.isArray(obj)) {
+              // Array elements will contain many integer values
+              // if we return them in memberGroups they will appear
+              // as obj.0 obj.100 etc. which is not valid syntax.
+              // Additionally for large arrays this can make for a
+              // very cluttered list.  So I filter them out.
+              memberGroups
+                .push(Object
+                      .getOwnPropertyNames(obj)
+                      .filter(function(item) {
+                        return isNaN(parseInt(item));
```

Comment:
Yeah, that's actually a 'feature' ;-) ... consider.. `var m = []; m['123abc'] = 'a'` .. if you do tab completion, the `123abc` and use `+item`, the `123abc` would still appear, which ends up having the same problem this patch is trying to fix. Using `parseInt` filters those out, even if they are valid keys for the object.